pkg/virt-api/rest/subresource.go

```diff
@@ -107,49 +96,141 @@ func (app *SubresourceAPIApp) streamRequestHandler(request *restful.Request, res
 	defer clientSocket.Close()
 
 	log.Log.Object(vmi).Infof("Websocket connection upgraded")
-	wsReadWriter := &kubecli.BinaryReadWriter{Conn: clientSocket}
 
-	inReader, inWriter := io.Pipe()
-	outReader, outWriter := io.Pipe()
+	tlsConfig, err := app.getConsoleTLSConfig()
```

Comment:
It looks like this function is doing a "get" on the virt-handler secret for every request. We don't have cert rotation right now, so we really only need to to pull in the secret once.  Once cert rotation comes into play, we'll have some sort of shared code that ensures the tls configs are always up to date with the latest.

@rmohr keep me honest here though, does what I'm saying sound correct based on what you have planned?

Reply:
Ack, makes sense